### PR TITLE
convert Interop to Truffle Libraries

### PR DESCRIPTION
**Pull request by lukasstadler**
These changes mainly update the interop code in grCUDA (in DeviceArray, etc.) to the new interop API in Truffle.
Additionally, there's some other minor updates:
- Modify the makefile that runs the parser generator to generate only the necessary files (no listeners, etc.) and to pre-process all files at once (fixes build problems).
- Don't store the context in the language, and use `@CachedContext(GrCUDALanguage.class)` in the right places.

### CODE REVIEW

File: projects/com.nvidia.grcuda/src/com/nvidia/grcuda/functions/DeviceArrayFunction.java [hunk 1]
```diff
@@ -46,17 +50,17 @@ public DeviceArrayFunction(CUDARuntime runtime) {
...
-        Object[] arguments = frame.getArguments();
-        if (arguments.length < 3) {   // arg 0 is the function itself
-            throw new RuntimeException("DeviceArray function expects at least two arguments.");
+    @TruffleBoundary
+    public Object call(Object[] arguments) throws ArityException, UnsupportedTypeException {
+        if (arguments.length < 2) {
+            throw ArityException.create(2, arguments.length);
         }
-        String typeName = expectString(arguments[1], "argument 1 of DeviceArray must be string (type name)");
+        String typeName = expectString(arguments[1], "first argument of DeviceArray must be string (type name)");
```
Comment (muellren) [inline projects/com.nvidia.grcuda/src/com/nvidia/grcuda/functions/DeviceArrayFunction.java]: BUG: Must be arguments[0] instead of argument one.

**Review by muellren — CHANGES_REQUESTED** [review]
There is one bug in DeviceArrayFunction. There is a new DeviceArrayTest Unit test I plan to merge soon.

File: projects/com.nvidia.grcuda/src/com/nvidia/grcuda/functions/DeviceArrayFunction.java (hunk 1 above)
Comment (lukasstadler) [reply projects/com.nvidia.grcuda/src/com/nvidia/grcuda/functions/DeviceArrayFunction.java]: ah, of course - I had that in a separate commit that I forgot to cherry pick.

**Review by muellren — APPROVED** [review]
Thanks, these changes look good to me. The polyglot language tests are passing as well. 
Go ahead, you should be able to merge the PR if this is setup correctly.

**Pull request merged**